aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/EndpointBuiltInsDecorator.kt

```diff
@@ -0,0 +1,179 @@
+    // load the builtIn with a matching name from the ruleset allowing for any docs updates
+    return rules.getBuiltIn(builtInSrc)
+}
+
+fun Model.sdkConfigSetter(
+    serviceId: ShapeId,
+    builtInSrc: Parameter,
+    configParameterNameOverride: String?,
+): Pair<AdHocSection<*>, (Section) -> Writable>? {
+    val builtIn = loadBuiltIn(serviceId, builtInSrc) ?: return null
+    val fieldName = configParameterNameOverride ?: builtIn.name.rustName()
+
+    val map = when (builtIn.type!!) {
+        ParameterType.STRING -> writable { rust("|s|s.to_string()") }
+        ParameterType.BOOLEAN -> null
+    }
+    return SdkConfigSection.copyField(fieldName, map)
+}
+
+/**
+ * Create a client codegen decorator that creates bindings for a builtIn parameter. Optionally, you can provide [clientParam]
+ * which allows control over the config parameter that will be generated.
+ */
+fun decoratorForBuiltIn(
+    builtIn: Parameter,
+    clientParam: ConfigParam? = null,
+): ClientCodegenDecorator {
+    val nameOverride = clientParam?.name
+    val name = nameOverride ?: builtIn.name.rustName()
+    return object : ClientCodegenDecorator {
+        override val name: String = "Auto${builtIn.builtIn.get()}"
+        override val order: Byte = 0
+
+        private fun rulesetContainsBuiltIn(codegenContext: ClientCodegenContext) =
+            codegenContext.getBuiltIn(builtIn) != null
+
+        override fun extraSections(codegenContext: ClientCodegenContext): List<Pair<AdHocSection<*>, (Section) -> Writable>> {
+            return listOfNotNull(
+                codegenContext.model.sdkConfigSetter(codegenContext.serviceShape.id, builtIn, clientParam?.name),
+            )
+        }
+
+        override fun configCustomizations(
+            codegenContext: ClientCodegenContext,
+            baseCustomizations: List<ConfigCustomization>,
+        ): List<ConfigCustomization> {
+            return baseCustomizations.extendIf(rulesetContainsBuiltIn(codegenContext)) {
+                standardConfigParam(
+                    clientParam ?: toConfigParam(builtIn),
+                )
+            }
+        }
+
+        override fun endpointCustomizations(codegenContext: ClientCodegenContext): List<EndpointCustomization> = listOf(
+            object : EndpointCustomization {
+                override fun loadBuiltInFromServiceConfig(parameter: Parameter, configRef: String): Writable? =
+                    when (parameter.builtIn) {
+                        builtIn.builtIn -> writable {
+                            rust("$configRef.$name")
+                            if (parameter.type == ParameterType.STRING) {
+                                rust(".clone()")
+                            }
+                        }
+                        else -> null
+                    }
+
+                override fun setBuiltInOnServiceConfig(name: String, value: Node, configBuilderRef: String): Writable? {
```

Comment:
We should consider thinking of a word that means "this function constructs and returns a writable" and using it to name things like this. My reasoning is that this looks like something that sets a field in Kotlin when it actually creates rust code that sets something in Rust. I don't think we have too much of a problem with keeping Kotlin vs. Rust stuff straight in codegen, but I think it's worth considering and defining a boundary.